Knapsack/src/knapsack.cpp

```diff
@@ -0,0 +1,134 @@
+// Copyright 2020 Ragimov Islam, Pogotovsky Pavel
+
+#include <vector>
+#include <queue>
+#include <algorithm>
+
+#include "../includes/knapsack.h"
+
+namespace np {
+
+size_t Knapsack::BruteForce() {
+    size_t num_of_cases = std::pow(2, elements_.size());
+    size_t count = 0, sol_count = 0;
+    size_t total_weight = 0, total_profit = 0;
+    std::vector<int> max_set_buf;
+    max_set_buf.resize(elements_.size());
+
+    for (size_t k = 0; k < num_of_cases; k++) {
+        for (size_t j = 0; j < elements_.size(); j++) {
+            if ((1 << j) & k) {
+                total_profit = total_profit + elements_[j].profit;
+                total_weight = total_weight + elements_[j].weight;
```

Comment:
It would be less work if break the loop by the if expression **if (total_weight < capacity_)**